Add cmake build configuration

p12tic opened the pull request:
cc @simogasp

simogasp commented on CMakeLists.txt:
```diff
@@ -0,0 +1,109 @@
+
+cmake_minimum_required(VERSION 3.11)
+
+project(CoinUtils LANGUAGES C CXX)
+
+set(PROJECT_VERSION 2.10.14)
+set(PROJECT_VERSION_MAJOR 2)
+set(PROJECT_VERSION_MINOR 10)
+set(PROJECT_VERSION_PATCH 14)
```
```suggestion
project(CoinUtils LANGUAGES C CXX VERSION 2.10.14)
```

p12tic replied:
Fixed now.

simogasp commented on CMakeLists.txt:
```diff
@@ -0,0 +1,109 @@
+    message(FATAL_ERROR "Missing necessary include files, like cmath or cfloat")
+endif()
+
+file(GLOB UTILS_SOURCES
```
same comment as the other repos, it's up to you if you want to do it

p12tic replied:
Fixed now.

simogasp commented on CMakeLists.txt:
```diff
@@ -0,0 +1,109 @@
+set(INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+
+if(MSVC)
+  set(
+    CMAKE_CXX_FLAGS
+    "${CMAKE_CXX_FLAGS} /bigobj /MP /wd4996 /wd4819 /wd4251 /wd4267 /wd4244 /wd4275"
```
nitpicking, I'd use `target_compile_options` and move this after the declaration of the target
```suggestion
  target_compile_options(${PROJECT_NAME} PRIVATE /bigobj /MP /wd4996 /wd4819 /wd4251 /wd4267 /wd4244 /wd4275)
```
and the same for the APPLE below.
But it's really if you are up to it, the current code does the job too

p12tic replied:
Fixed now.

p12tic committed:
Add cmake configuration from vcpkg

The cmake configuration has been taken from vcpkg commit
207fbcb55cef3544f3314ef783d169419f251b3f.

p12tic committed:
Fix paths in CMake configuration

p12tic committed:
Use standard install paths for CMake files

p12tic committed:
Cleanup project definition for CMake

p12tic committed:
Don't use GLOB to get list of files to compile in CMake

p12tic committed:
Clean up compile options setup in CMake

p12tic commented:
@simogasp I've addressed your comments.

simogasp reviewed (approved):
Thanks!
